File: ios/EMWaverTests/ActiveDeviceTargetTests.swift
import XCTest
@testable import EMWaver

final class ActiveDeviceTargetTests: XCTestCase {
    private enum Transport {
        case none
        case usb
        case ble
    }

    func testNormalizesBlankDeviceIdToActive() {
        let target = ActiveDeviceTarget(deviceId: "   ", transport: Transport.none)

        XCTAssertEqual(target.deviceId, "active")
        XCTAssertTrue(target.matchesDeviceId(nil))
        XCTAssertTrue(target.matchesDeviceId(""))
    }

    func testMatchesExactTrimmedDeviceIdAndTransport() {
        let target = ActiveDeviceTarget(deviceId: " usb:board-1 ", transport: Transport.usb)

        XCTAssertEqual(target.deviceId, "usb:board-1")
        XCTAssertTrue(target.matchesDeviceId("usb:board-1"))
        XCTAssertFalse(target.matchesDeviceId("ble:board-1"))
        XCTAssertTrue(target.matchesTransport(.usb))
        XCTAssertFalse(target.matchesTransport(.ble))
    }
}

File: windows/EMWaver.Tests/ActiveDeviceTargetTests.cs
using EMWaver.Services;
using Xunit;

namespace EMWaver.Tests;

public sealed class ActiveDeviceTargetTests
{
    [Fact]
    public void NormalizesBlankDeviceIdToActive()
    {
        var target = new ActiveDeviceTarget("   ", DeviceTransport.None);

        Assert.Equal("active", target.DeviceId);
        Assert.True(target.MatchesDeviceId(null));
        Assert.True(target.MatchesDeviceId(""));
    }

    [Fact]
    public void MatchesTrimmedDeviceIdCaseInsensitivelyAndTransport()
    {
        var target = new ActiveDeviceTarget(" usb:Board-1 ", DeviceTransport.UsbMidi);

        Assert.Equal("usb:Board-1", target.DeviceId);
        Assert.True(target.MatchesDeviceId("USB:board-1"));
        Assert.False(target.MatchesDeviceId("ble:board-1"));
        Assert.True(target.MatchesTransport(DeviceTransport.UsbMidi));
        Assert.False(target.MatchesTransport(DeviceTransport.Ble));
    }
}
